### PR TITLE
backend/cuda: Fix bashism in shell script

### LINKED ISSUE #180: bash syntax in sh script
Hi folks,

https://github.com/pmodels/yaksa/blob/957ff4fc24de2eb87c8892cbd3ec41759c965848/src/backend/cuda/cudalt.sh#L35

The bash substring expansion `${@:2}` is an extension that's not present in all `/bin/sh` implementations.  It caused a `bad substitution` error when I was configuring on a system where `/bin/sh` is dash.

I apologize for not submitting a patch: I didn't try to understand the line that caused the error, I just changed the shebang to `#!/bin/bash` and I thought you wouldn't want to do that for a real fix.

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Use shift instead of array slicing syntax. Fixes a build issue on
systems with dash as /bin/sh, e.g. Ubuntu. Closes #180.

## Expected Impact

Fix build issue on Ubuntu.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Have read and agree to the Yaksa CLA terms (https://github.com/pmodels/yaksa/wiki/Yaksa-Contributor-License-Agreement)

### CODE REVIEW

**Comment by hzhou** [comment]
test:yaksa/gpu

File: src/backend/cuda/cudalt.sh [hunk 1]
```diff
@@ -32,11 +32,12 @@ if test ! -d "$PIC_DIR" ; then
     mkdir -p "$PIC_DIR"
 fi
 
-CMD="${@:2} -Xcompiler -fPIC -o $PIC_FILEPATH"
+shift
```
Comment (hzhou) [inline src/backend/cuda/cudalt.sh:35]: Could you add a comment on what is being shifted? Maybe it is easier to read if we do the shift right after line 15?

**Review by hzhou — APPROVED** [review]
A minor comment. I leave it up to you.

File: src/backend/cuda/cudalt.sh (hunk 1 above)
Comment (raffenet) [reply src/backend/cuda/cudalt.sh:35]: Sure, can do.

**Comment by raffenet** [comment]
test:yaksa/gpu

**Pull request merged**